### PR TITLE
feat: cbor support no_std

### PR DESCRIPTION
**Pull request by NanYeZhuYu**

### CODE REVIEW

File: libs/ur-registry/src/extend/crypto_multi_accounts.rs [hunk 1]
```diff
@@ -67,83 +72,83 @@ impl RegistryItem for CryptoMultiAccounts {
...
     }
+}
 
-    fn from_bytes(bytes: Vec<u8>) -> Result<CryptoMultiAccounts, String> {
-        let value: Value = match from_slice(bytes.as_slice()) {
-            Ok(x) => x,
-            Err(e) => return Err(e.to_string()),
-        };
-        CryptoMultiAccounts::from_cbor(value)
+impl FromCbor<CryptoMultiAccounts> for CryptoMultiAccounts {
```
Comment (renfengshi) [inline libs/ur-registry/src/extend/crypto_multi_accounts.rs:141]: From

File: libs/ur-registry/src/solana/sol_sign_request.rs [hunk 2]
```diff
@@ -120,92 +139,131 @@ impl RegistryItem for SolSignRequest {
...
     }
+}
 
-    fn from_bytes(bytes: Vec<u8>) -> Result<SolSignRequest, String> {
-        let value: Value = match from_slice(bytes.as_slice()) {
-            Ok(x) => x,
-            Err(e) => return Err(e.to_string()),
-        };
-        SolSignRequest::from_cbor(value)
+impl FromCbor<SolSignRequest> for SolSignRequest {
```
Comment (renfengshi) [inline libs/ur-registry/src/solana/sol_sign_request.rs:223]: From

File: libs/ur-registry/src/solana/sol_signature.rs [hunk 3]
```diff
@@ -44,51 +49,86 @@ impl RegistryItem for SolSignature {
...
     }
+}
 
-    fn from_bytes(bytes: Vec<u8>) -> Result<SolSignature, String> {
-        let value: Value = match from_slice(bytes.as_slice()) {
-            Ok(x) => x,
-            Err(e) => return Err(e.to_string()),
-        };
-        SolSignature::from_cbor(value)
+impl FromCbor<SolSignature> for SolSignature {
```
Comment (renfengshi) [inline libs/ur-registry/src/solana/sol_signature.rs:100]: From

File: libs/ur-registry/src/crypto_output.rs [hunk 4]
```diff
@@ -32,33 +57,144 @@ impl RegistryItem for CryptoOutput {
...
+    fn to_bytes(&self) -> URResult<Vec<u8>> {
+        minicbor::to_vec(self.clone()).map_err(|e| URError::CborEncodeError(e.to_string()))
     }
+}
 
-    fn to_bytes(&self) -> Vec<u8> {
-        let value = self.to_cbor();
-        //TODO: remove unwrap
-        to_vec(&value).unwrap()
+impl FromCbor<CryptoOutput> for CryptoOutput {
```
Comment (renfengshi) [inline libs/ur-registry/src/crypto_output.rs:119]: From

File: libs/ur-registry/src/extend/crypto_multi_accounts.rs (hunk 1 above)
Comment (NanYeZhuYu) [reply libs/ur-registry/src/extend/crypto_multi_accounts.rs:141]: > From

跟别的trait同名，所以通过From as FromCbor  rename了一下

**Review by renfengshi — APPROVED** [review]

**Pull request merged**